api/auth_middleware_test.go

```diff
@@ -1,63 +1,101 @@
 package api_test
 
 import (
+	b64 "encoding/base64"
 	"fmt"
 	"net/http"
 	"testing"
 
+	"github.com/topfreegames/Will.IAM/models"
 	helpers "github.com/topfreegames/Will.IAM/testing"
 )
 
-func TestAuthMiddlewareKeyPairShouldAuthenticateUser(t *testing.T) {
+func TestAuthMiddleware(t *testing.T) {
 	helpers.CleanupPG(t)
 
-	rootSA := helpers.CreateRootServiceAccountWithKeyPair(t)
-	app := helpers.GetApp(t)
-	req, err := http.NewRequest(http.MethodGet, "/service_accounts", nil)
-
-	if err != nil {
-		t.Errorf("Could not create HTTP request for /service_accounts")
-	}
-
-	req.Header.Set("Authorization", fmt.Sprintf(
-		"keypair %s:%s", rootSA.KeyID, rootSA.KeySecret,
-	))
-
-	response := helpers.DoRequest(t, req, app.GetRouter())
-
-	if response.Code != http.StatusOK {
-		t.Errorf("Expected status %d. Got %d", http.StatusOK, response.Code)
-	}
+	oauthSA := helpers.CreateRootServiceAccountWithOAuth(t, "oauthUser", "oauth.user@test.com")
+	keyPairSA := helpers.CreateRootServiceAccountWithKeyPair(t, "keyPairUser", "keypair.user@test.com")
+	tokensRepo := helpers.GetRepo(t).Tokens
+	tokens, _ := tokensRepo.FindByEmail(oauthSA.Email)
+	token := tokens[0]
+	emptyMap := make(map[string]string)
```

Comment:
nit: kill this and inline `map[string]string{}` where it's used? We don't need to alias an empty map if we can just place an actual empty map in its place.

Reply:
My first thought was to do that, but I ended up creating a variable for that to make it clear that we are passing an empty Map to the tests - give it a name instead of infer that its an empty Map. WDYT?

Reply:
I'd rather use `map[string]string{}`, it's clear enough and one less variable name to think about.

Better yet, just don't set requestHeaders/expectedResponseHeaders for those tests - they should just fine. No need for either variable nor map[string]string{}.